src/Ocelot.Provider.Polly/v7/PollyQoSProvider.cs

```diff
@@ -60,12 +57,10 @@ private PollyPolicyWrapper<HttpResponseMessage> PollyPolicyWrapperFactory(Downst
 
         // No default set for polly timeout at the minute.
         // Since a user could potentially set timeout value = 0, we need to handle this case.
-        // TODO throw an exception if the user sets timeout value = 0 or at least return a warning
-        // TODO the design in DelegatingHandlerHandlerFactory should be reviewed
         var timeoutPolicy = Policy
-            .TimeoutAsync<HttpResponseMessage>(
-                TimeSpan.FromMilliseconds(route.QosOptions.TimeoutValue),
-                TimeoutStrategy.Pessimistic);
+            .TimeoutAsync<HttpResponseMessage>(route.QosOptions.TimeoutValue > 0
```

Comment:
useless, V7 will be remove in the next release 24, so, It will be better to change nothing to the behavior

Reply:
Fixed in commit https://github.com/ThreeMammals/Ocelot/pull/2073/commits/e87010a9d36462e65dc06d358781f26545bc6898